kubetest2-gke/deployer/build.go

```diff
@@ -47,3 +52,31 @@ func (d *deployer) verifyBuildFlags() error {
 	d.BuildOptions.RepoRoot = d.RepoRoot
 	return d.BuildOptions.Validate()
 }
+
+// ensure that the version is a valid gke version
+func normalizeVersion(version string) (string, error) {
+	// (1.18.10)(-gke)(.601.123)(+existingSuffix)
+	// capture repeated occurrences of group 3 and don't capture just last
+	re, err := regexp.Compile(`(\d\.\d+\.\d*)(-\w+)?((?:\.\d+)*)(\+.*)*`)
+	if err != nil {
+		return "", err
+	}
+	matches := re.FindStringSubmatch(version)
+	if matches == nil || len(matches) != 5 {
+		return "", fmt.Errorf("%q is not a valid gke version", version)
```

Comment:
include matches in error for debugging?
```suggestion
		return "", fmt.Errorf("%q is not a valid gke version: %#v", version, matches)
```

Reply:
done